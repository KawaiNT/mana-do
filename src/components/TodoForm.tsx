import React, { useRef} from 'react';
import { TodoFormInterface } from '../models/todo';


const TodoForm = (props: TodoFormInterface) => {
  const inputRef = useRef<HTMLInputElement>(null);

  const onKeyDown = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    // Make sure have value before creating a todo
    if (e.key === 'Enter' && inputRef.current && inputRef.current.value) {
      props.onKeyDown(inputRef.current.value)
      inputRef.current.value = '';
    }
  }
  
  return (
    <div className="Todo__creation" data-testid="todo-form">
      <input
        ref={inputRef}
        className="Todo__input"
        placeholder="What need to be done?"
        onKeyDown={onKeyDown}
      />
    </div>
  )
}

export default TodoForm