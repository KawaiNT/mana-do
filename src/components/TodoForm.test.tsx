import React from 'react';
import { render } from '@testing-library/react';
import TodoForm from './TodoForm'

describe('TodoForm component test', () => {
  test('TodoForm render', () => {
    const onKeyDown = jest.fn()
    const { asFragment } = render(<TodoForm onKeyDown={onKeyDown} />)

    expect(asFragment(<TodoForm onKeyDown={onKeyDown} />)).toMatchSnapshot()
  })

  test('TodoForm should be in the document', () => {
    const onKeyDown = jest.fn()
    const { getByTestId } = render(<TodoForm onKeyDown={onKeyDown} />)
    const todoForm = getByTestId('todo-form')

    expect(todoForm).toBeInTheDocument()
  })
})