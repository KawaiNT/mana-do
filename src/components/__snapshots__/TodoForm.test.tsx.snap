// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm component test TodoForm render 1`] = `
<DocumentFragment>
  <div
    class="Todo__creation"
    data-testid="todo-form"
  >
    <input
      class="Todo__input"
      placeholder="What need to be done?"
    />
  </div>
</DocumentFragment>
`;
